staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go

```diff
@@ -105,7 +105,9 @@ func (a statusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Obj
 	var correlatedObject *common.CorrelatedObject
 	if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CRDValidationRatcheting) {
 		correlatedObject = common.NewCorrelatedObject(uNew.Object, uOld.Object, &model.Structural{Structural: a.structuralSchema})
-		options = append(options, validation.WithRatcheting(correlatedObject.Key("status")))
+		if a.structuralSchema != nil {
```

Comment:
does this protect against all of the following?
- the CRD not having a schema at all
- the CRD having a schema but it not being structural
- the CRD having a structural schema but not declaring the "status" property. I think a.statusSchemaValidator will be nil in this case (https://github.com/kubernetes/kubernetes/blob/0f710d2738c32cc880cc8429cf25e5c7701b5619/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L792-L793)

Reply:
Only "the CRD having a schema but it not being structural" will cause this issue. In my test, Other two case is ok and does not need this commit.